### PR TITLE
[10.0] [ADD] hr_timesheet_sheet_period

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
HR Timesheet Sheet based on Payroll Period
==========================================

This module allows to create timesheets with start and end dates matching with the
payroll period of the schedule pay that is specified in the contract of the 
employee or if not then will match the nearest payroll period.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review code.

**Comment by LoisRForgeFlow** [comment]
Depens on https://github.com/OCA/hr/pull/352

**Comment by LoisRForgeFlow** [comment]
Can you please attend travis? @nikul-serpentcs

**Comment by nikul-serpentcs** [comment]
@lreficent Because depends on module `hr_period`, is not available.

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            return period.date_start
+        else:
+            return res
+
+    @api.model
+    def _default_date_to(self):
+        res = super(HrTimesheetSheet, self)._default_date_to()
+        period = self._get_current_pay_period()
+        if period:
+            return period.date_stop
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: The field in the date_range module and in the hr_period module is `date_end`

File: hr_timesheet_sheet_period/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    @api.model
+    def _get_current_pay_period(self):
+        period_obj = self.env['hr.period']
+        contract_obj = self.env['hr.contract']
+        date_today = datetime.today().strftime('%Y-%m-%d')
+        employee = self.default_get(['employee_id'])
+        contract = contract_obj.search([('employee_id', '=',
+                                         employee.get('employee_id'))])
+        search_domain = [('date_start', '<=', date_today),
+                         ('date_stop', '>=', date_today)]
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_period/models/hr_timesheet_sheet.py]: date_end

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]

**Comment by nikul-serpentcs** [comment]
@aheficent I did rename a fields, Could you please review code.

**Comment by AaronHForgeFlow** [comment]
Hi @nikul-serpentcs  Thank you for attending the commits. Can you review and merge this? https://github.com/nikul-serpentcs/hr-timesheet/pull/1

**Comment by AaronHForgeFlow** [comment]
Hi @nikul-serpentcs I'll take care of this PR here: https://github.com/OCA/hr-timesheet/pull/123 Feel free to close this

**Pull request closed without merging**